### PR TITLE
Rake

### PR DESCRIPTION
**Pull request by Will-D-M**
- Created model tests

- Created models

- Created rake import

### CODE REVIEW

File: lib/tasks/import.rake [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    CSV.foreach(file, headers: true) do |row|
+      invoice_item_hash = row.to_hash
+      invoice_item = InvoiceItem.where(id: invoice_item_hash["id"])
+      if invoice_item.count == 1
+        invoice_item.first.update_attributes(invoice_item_hash)
+      else
+        InvoiceItem.create!(invoice_item_hash)
+      end
+  end
+end
```
Comment (Will-D-M) [inline lib/tasks/import.rake]: @BrianZanti @dionew1

This seems kind of messy, but I spent hours on it, and it's working in dev, so I am going to roll with it. 

I essentially found this online and I understand how it is pulling from the CSV, but I am unfamiliar with the update_attributes action and why exactly that if statement needs to be there. Can either of you explain what is occurring within that if-statement?

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  #
+  # The different available types are documented in the features, such as in
+  # https://relishapp.com/rspec/rspec-rails/docs
+  config.infer_spec_type_from_file_location!
+
+  # Filter lines from Rails gems in backtraces.
+  config.filter_rails_from_backtrace!
+  # arbitrary gems may also be filtered via:
+  # config.filter_gems_from_backtrace("gem name")
+end
```
Comment (Will-D-M) [inline spec/rails_helper.rb]: @dionew1 @brianzanti

I am also having a little trouble with my shoulda matchers. I was messing around quite a bit with my ruby versions during setup, and I think I may have tweaked something. When I run my rspec for my model tests, I am getting an error of "undefined method belong_to." Any idea what might be occurring there or why it's not working? I did add shoulda matchers to the gemfile and installed.

File: lib/tasks/import.rake (hunk 1 above)
Comment (dionew1) [reply lib/tasks/import.rake]: That's a great question. I'm going to talk to the Invoice Item specifically, but it is what is happening generally with all of them. The line directly before the `if` block is `invoice_item = InvoiceItem.where(id: invoice_item_hash["id"])`.  The variable `invoice_item` is being set to an array of all the invoice items that have a matching id from the current row that you are on.
Now the 'if' block is saying that "if" an invoice item already exists with that id then we are updating.  The else is creating the Invoice Item if it doesn't exist.

In summary, it just a way to handle if there were duplicate ids within the CSV file.

File: spec/rails_helper.rb (hunk 2 above)
Comment (dionew1) [reply spec/rails_helper.rb]: At first glance, it looks like you are missing this 

`Shoulda::Matchers.configure do |config|
  config.integrate do |with|
    with.test_framework :rspec
    with.library :rails
  end
end`

I would make sure this is after the `end` on line 63. I believe Cassie might have been having a similar issue and Brian and she were able to come to a solution, so if you still have issues I might reach out to her or Brian.

Comment (dionew1) [reply spec/rails_helper.rb]: I would look at the docs [here](https://github.com/thoughtbot/shoulda-matchers) since the code block I included didn't format too well.
